logical/plugin/system.go

```diff
@@ -0,0 +1,245 @@
+package plugin
+
+import (
+	"net/rpc"
+	"time"
+
+	"fmt"
+
+	plugin "github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/vault/helper/consts"
+	"github.com/hashicorp/vault/helper/pluginutil"
+	"github.com/hashicorp/vault/helper/wrapping"
+	"github.com/hashicorp/vault/logical"
+)
+
+type SystemViewClient struct {
+	client *rpc.Client
+}
+
+func (s *SystemViewClient) DefaultLeaseTTL() time.Duration {
+	var reply DefaultLeaseTTLReply
+	err := s.client.Call("Plugin.DefaultLeaseTTL", new(interface{}), &reply)
+	if err != nil {
+		return 0
+	}
+
+	return reply.DefaultLeaseTTL
+}
+
+func (s *SystemViewClient) MaxLeaseTTL() time.Duration {
+	var reply MaxLeaseTTLReply
+	err := s.client.Call("Plugin.MaxLeaseTTL", new(interface{}), &reply)
+	if err != nil {
+		return 0
+	}
+
+	return reply.MaxLeaseTTL
+}
+
+func (s *SystemViewClient) SudoPrivilege(path string, token string) bool {
+	var reply SudoPrivilegeReply
+	args := &SudoPrivilegeArgs{
+		Path:  path,
+		Token: token,
+	}
+
+	err := s.client.Call("Plugin.SudoPrivilege", args, &reply)
+	if err != nil {
+		return false
+	}
+
+	return reply.Sudo
+}
+
+func (s *SystemViewClient) Tainted() bool {
+	var reply TaintedReply
+
+	err := s.client.Call("Plugin.Tainted", new(interface{}), &reply)
+	if err != nil {
+		return false
+	}
+
+	return reply.Tainted
+}
+
+func (s *SystemViewClient) CachingDisabled() bool {
+	var reply CachingDisabledReply
+
+	err := s.client.Call("Plugin.CachingDisabled", new(interface{}), &reply)
+	if err != nil {
+		return false
+	}
+
+	return reply.CachingDisabled
+}
+
+func (s *SystemViewClient) ReplicationState() consts.ReplicationState {
+	var reply ReplicationStateReply
+
+	err := s.client.Call("Plugin.ReplicationState", new(interface{}), &reply)
+	if err != nil {
+		return consts.ReplicationDisabled
+	}
+
+	return reply.ReplicationState
+}
+
+func (s *SystemViewClient) ResponseWrapData(data map[string]interface{}, ttl time.Duration, jwt bool) (*wrapping.ResponseWrapInfo, error) {
+	var reply ResponseWrapDataReply
+	args := &ResponseWrapDataArgs{
+		Data: data,
+		TTL:  ttl,
+		JWT:  jwt,
```

Comment:
As a hint, it was for plugin TLS wrapping. Was it because it can't unwrap the TLS token over the plugin connection, so it needed to have the API address in the JWT encoding?